motionplan/plannerOptions.go

```diff
@@ -102,9 +103,10 @@ func newBasicPlannerOptions(frame referenceframe.Frame) *plannerOptions {
 // plannerOptions are a set of options to be passed to a planner which will specify how to solve a motion planning problem.
 type plannerOptions struct {
 	ConstraintHandler
-	goalMetric   ik.StateMetric // Distance function which converges to the final goal position
-	goalArcScore ik.SegmentMetric
-	pathMetric   ik.StateMetric // Distance function which converges on the valid manifold of intermediate path states
+	goalMetricConstructor func(spatialmath.Pose) ik.StateMetric
```

Comment:
This is necessary in order to be able to set a goal from just a pose. Either we need to do this, or we need to update `plan` to accept a metric rather than a pose.